src/android/CameraPreviewFragment.java

```diff
@@ -124,31 +146,41 @@ public void startCamera() {
             targetResolution = CameraPreviewFragment.calculateResolution(getContext(), targetSize);
         }
 
+        Recorder recorder = new Recorder.Builder()
+                .setQualitySelector(QualitySelector.from(Quality.LOWEST))
+                .build();
+        videoCapture = VideoCapture.withOutput(recorder);
+
+
         preview = new Preview.Builder().build();
         imageCapture = new ImageCapture.Builder()
                 .setTargetResolution(targetResolution)
                 .build();
-
-        cameraProvider.unbindAll();
-        try {
-            camera = cameraProvider.bindToLifecycle(
-                    this,
-                    cameraSelector,
-                    preview,
-                    imageCapture
-            );
-        } catch (IllegalArgumentException e) {
-            // Error with result in capturing image with default resolution
-            e.printStackTrace();
+        this.getActivity().runOnUiThread(() -> {
```

Comment:
```suggestion
        this.getActivity().runOnUiThread(() -> {
```

Why do we need to run this on UI thread?